dso/dependabot.py

```diff
@@ -0,0 +1,187 @@
+    clogger = dso.util.component_logger(__name__)
+    DEPENDABOT_YAML_PATHS = (
+        '.dependabot.yaml',
+        '.dependabot.yml',
+        '.github/.dependabot.yaml',
+        '.github/.dependabot.yml',
+        'dependabot.yaml',
+        'dependabot.yml',
+        '.github/dependabot.yaml',
+        '.github/dependabot.yml'
+    )
+
+    repo_dir = ci.util.existing_dir(pathlib.Path(repo_dir))
+    if not repo_dir.joinpath('.git').is_dir():
+        raise ValueError(f'not a git root directory: {repo_dir}')
+
+    for path in DEPENDABOT_YAML_PATHS:
+        dependabot_yaml = repo_dir.joinpath(path)
+        if dependabot_yaml.is_file():
+            return True
+
+    clogger.info('dependabot not found in any known places')
+    return False
+
+
+def is_dependabot_in_component_repo(
+    repo: str,
+    github: str,
+    github_cfg
+) -> bool:
+    clogger = dso.util.component_logger(__name__)
+
+    clogger.info(f'cloning {github}/{repo}')
+    with tempfile.TemporaryDirectory() as temp_dir:
+        gitutil.GitHelper.clone_into(
+            target_directory=temp_dir,
+            github_cfg=github_cfg,
+            github_repo_path=repo,
+        )
+
+        return _dependabot_yaml_in_known_places(repo_dir=temp_dir)
+
+
+def _get_repo_in_org(org_name: str):
+    clogger = dso.util.component_logger(__name__)
+    host, org = org_name.split('/')
+    clogger.info(f'looking for repos from {org=} on {host=}')
+
+    github_cfg = ccc.github.github_cfg_for_hostname(
+        cfg_factory=ci.util.ctx().cfg_factory(),
+        host_name=host,
+    )
+    github_api = ccc.github.github_api(github_cfg)
+    github_org = github_api.organization(org)
+    return github_org.repositories
+
+
+def _scan_repos_for_dependabot(
+    repos: typing.List[str],
+    host: str,
+) -> dso.model.DependabotCoverageReport:
+
+    clogger = dso.util.component_logger(__name__)
+    i = 1
+
+    report: dso.model.DependabotCoverageReport
+    details = []
+
+    clogger.info(f'scanning {len(repos)} repos')
+    for repo in repos:
+        github_cfg = ccc.github.github_cfg_for_hostname(
+            cfg_factory=ci.util.ctx().cfg_factory(),
+            host_name=host,
+        )
+        clogger.info(f'{i} / {len(repos)}')
+        dependabot_found = is_dependabot_in_component_repo(
+            repo=repo,
+            github_cfg=github_cfg,
+            github=host
+        )
+        repo_report = dso.model.DependabotCoverageReportRepo(
+            dependabot=dependabot_found,
+            repo=repo
+        )
+        details.append(repo_report)
+        i += 1
+
+    return dso.model.DependabotCoverageReport(
+        coverage=0,
+        details=details,
+        github=host,
+    )
+
+
+def _get_github_api(host):
+    github_cfg = ccc.github.github_cfg_for_hostname(
+        cfg_factory=ci.util.ctx().cfg_factory(),
+        host_name=host,
+    )
+    return ccc.github.github_api(github_cfg)
+
+
+def _get_scoped_repos_for_org(
```

Comment:
I do not think `scoped` is an appropriate term(*). In GitHub's terms, I "my" terms (as used elsewhere in cc-utils), I would tend to call `{org}/{repo}` a repo(sitory)-path. I think, github's API calls this the `full_name` of a repository. So repository-name might also be  adequate. or maybe qualified name.

(*) one could argue that the repo-name (w/o org) is only valid in the scope of its organisation. However, at least from my experience, I never saw the naming convention of calling a qualified name a scoped name (as opposed to canonical, qualified, or full name), and I find it unintuitive, at least (maybe it is not strictly "wrong").